### ISSUE TITLE
[Doc]: Add visual key to user guide

### ISSUE THREAD

**Issue by story645** [matplotlib/matplotlib#26205]
### Documentation Link

https://matplotlib.org/devdocs/users/index.html

### Problem

It's hard to navigate the user guide if you don't know what we call things. Folks should start on the quick start so that they will learn what it's called, but a) we don't indicate that quickstart teaches that indormation, b) a lot of folks don't and we should account for that.  attn: @melissawm and @esibinga 

### Suggested improvement

I was thinking like pull out and streamline the [parts of figure](file:///C:/Users/story/Projects/matplotlib/doc/build/html/users/explain/quick_start.html#parts-of-a-figure) part of the quickstart guide and make it a big card. Something like: 

<details open>
<summary> Screenshot: Sample layout</summary>
<img src="https://github.com/matplotlib/matplotlib/assets/1300499/890d1006-2433-4748-bf7c-bc9f73afd3a2">

</details>

If folks like this direction, it can definitely be cleaned up to something pretty and probably slim down the anatomy figure to just the basics. Maybe take the longer descriptions from anatomy and work them into their corresponding intros. 

 And probably streamlined to just the contents of the [userguide index](file:///C:/Users/story/Projects/matplotlib/doc/build/html/users/explain/index.html) if #26156 gets resolved.

**Comment by akshayamadhuri** [comment]
@story645  i would like to work on this

**Comment by jklymak** [comment]
If a new user is confused by what we mean by "figure" or "axes" those are pretty clearly listed in the TOC and explained in the Getting Started.  Im not a fan of redefining everything on every page a user may land on as that is distracting to the other 99.99% of people who have already read the Getting Started. If there are issues with the TOC we should address those directly.

**Comment by jklymak** [comment]
Sorry, confused Quick Start and Getting Started, but I think you get the idea.  

> Yes but if you don't know what they are than you don't know what that means in the TOC -> like you have to know "oh, Axes is the thing I make visuals on, let me look at the Axes" tutorial. Partially it's intended as motivation to read the other things.

If I go to a webpage, and it has a bunch of terminology in the TOC I don't understand, and it has something called "Getting Started" or "Quick Start", then I will read those first.  I don't know that I'd expect a glossary in the TOC, and I don't think I'd expect to comprehend every term in the TOC before I read those sections.   There is only so much bootstrapping you can do before things become very disorganized.

**Comment by jklymak** [comment]
I don't see that any of those have a glossary on the landing page, though I agree that Bokeh has one in their introduction (as do we!).  

> My use case is that I just want to know what the thing I need to modify is called so I can go to those specific docs -> and honestly like I don't think that's uncommon given how often I've had to link folks to "hey, did you read the getting started" docs?

We all try to jump around in doc pages for things we just want to use, but don't really want to learn. For me it's sphinx, but just because I don't understand many sphinx concepts doesn't mean the sphinx docs are bad.  

Anyway, I'm not in favour of duplicating the glossary/Parts of Figure image on the lading page.  If you want to _move_ it there, that is less objectionable, but I still think out of place.

**Comment by jklymak** [comment]
I dont agree that they are orienting their audience more than us. I'd be fine improving Getting Started and Quick Start and maybe merging them. That is where I would expect to see detailed orientation.  I would not expect a glossary in the table of contents.

**Comment by jklymak** [comment]
It is absolutely our table of contents. Read conf.py.  There is discussion of reorganizing where the TOC is in another issue.

**Comment by jklymak** [comment]
It is absolutely our table of contents. Read conf.py.  There is discussion of reorganizing where the TOC is in another issue.

**Comment by jklymak** [comment]
If you are going to say this page should not be the table of contents then you need to say where the TOC should go, not just add extra info to the existing TOC. In my opinion the proposal here adds to the organization problem rather than helps solve it. 

I believe my stance has been consistent. We don't want or need a glossary in the table of contents.
